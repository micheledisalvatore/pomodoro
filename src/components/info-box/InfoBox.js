import React from 'react';
import PropTypes from 'prop-types';

import { Container, Current } from './InfoBox.styled';

export const InfoBox = ({ current, total, title }) => (
  <Container>
    <h3>{title}</h3>
    <Current>{current}</Current><span>/{total}</span>
  </Container>
);

InfoBox.propTypes = {
  title: PropTypes.string.isRequired,
  current: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
};
