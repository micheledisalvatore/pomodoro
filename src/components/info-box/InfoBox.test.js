import React from 'react';
import { shallow } from 'enzyme';

import { InfoBox } from './InfoBox';
import { Current } from './InfoBox.styled';

describe('Given a InfoBox component', () => {
  let component;
  let wrapper;

  const requiredProps = {
    title: 'foo',
    current: 3,
    total: 10,
  };

  describe('when it is rendered', () => {
    beforeEach(() => {
      component = <InfoBox {...requiredProps} />
      wrapper = shallow(component);
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })

    it('should show the given title', () => {
      expect(wrapper.find('h3')).toHaveText('foo');
    })

    it('should show the given current value', () => {
      expect(wrapper.find(Current)).toHaveText('3');
    })

    it('should show the given total value', () => {
      expect(wrapper.find('span')).toHaveText('/10');
    })
  })
})
