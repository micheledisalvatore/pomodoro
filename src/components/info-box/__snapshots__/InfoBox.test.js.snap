// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a InfoBox component when it is rendered should match the snapshot 1`] = `
<styled.div>
  <h3>
    foo
  </h3>
  <styled.span>
    3
  </styled.span>
  <span>
    /
    10
  </span>
</styled.div>
`;
